### PR TITLE
Generate and publish API docs

### LINKED ISSUE #53: Generate API documentation
### What's needed?

We need to generate the API documentation so it can be consulted more easily.

### Proposed solution

Use a tool to generate the API documentation by extracting it from the *docstrings*.

### Use cases

_No response_

### Alternatives and workarounds

Look at the documentation in the source code.

### Additional context

We need some particular requirements:

* We use [Google style documentation comments](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)
* We use markdown in documentation comments

Nice to haves:

* Support for [mermaid diagrams](https://mermaid-js.github.io/)

### PR DESCRIPTION
**Pull request by leandro-lucarella-frequenz**
The documentation is generated using MkDocs, using the Material theme plus mkdocstrings (and a few minor supporting extensions) to build the API documentation extracted from code comments.

### TODO

- [ ] Check if some more basic documentation fixing needs to be done.

#### Probably for another PR

- [ ] Think about what is the public interface, and hide modules that are internal. #73 
- [ ] Add class attributes documentation. #74
- [ ] Configure the default version in `mike` on the final website (after the release).

### Changes

- ci/labeler: Sort file patterns
- ci/labeler: Add missing noxfile.py
- Use MkDocs to build the documentation
- Improve mkdocstrings API generator
- Fix resampling actor documentation
- Don't generate documentation for internal modules
- Move copyright and license as code comments
- Improve documentation formatting for power distributor
- docs: Indent examples
- Use forward type annotations
- docs: Add site version via mike
- Mention the new website in RELEASE_NOTES
- ci: Build and publish documentation
- Run normal linting on noxfile.py
- Don't ignore imports when running mypy
- Add PyPI and Docs badges to README

### CODE REVIEW

**Comment by leandro-lucarella-frequenz** [comment]
You can see the results here: https://leandro-lucarella-frequenz.github.io/frequenz-sdk-python/next

File: noxfile.py [hunk 1]
```diff
@@ -46,9 +59,13 @@ def ci_checks_max(session: nox.Session) -> None:
...
 
     This does NOT run pytest_min, so that needs to be run separately as well.
+
+    Args:
+        session: the nox session.
     """
-    session.install(".[docs]", *PYTEST_DEPS, *FMT_DEPS, *LINT_DEPS,
-            *DOCSTRING_DEPS)
+    session.install(
+        ".[docs]", "mypy", "pylint", "nox", *PYTEST_DEPS, *FMT_DEPS, *DOCSTRING_DEPS
```
Comment (sahas-subramanian-frequenz) [inline noxfile.py]: why do we need `nox` inside a `nox` session?

**Review by sahas-subramanian-frequenz — APPROVED** [review]

File: noxfile.py (hunk 1 above)
Comment (leandro-lucarella-frequenz) [reply noxfile.py]: Yes. It installs a clean venv without `nox`.

Comment (shsms) [reply noxfile.py]: my question was "why?" :-)

**Pull request merged**